core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -63,6 +66,32 @@ class TopicConfigHandler(private val logManager: LogManager, kafkaConfig: KafkaC
       val logConfig = LogConfig(props)
       logs.foreach(_.config = logConfig)
     }
+
+    val brokerId = kafkaConfig.brokerId
+
+    if (topicConfig.containsKey(LogConfig.ThrottledReplicasListProp)) {
+      val partitions = parseThrottledPartitions(topicConfig, brokerId)
+      quotas.leader.markThrottled(topic, partitions)
+      quotas.follower.markThrottled(topic, partitions)
+      logger.info(s"Setting throttled partitions on broker $brokerId to $partitions")
+    }
+  }
+
+  def parseThrottledPartitions(topicConfig: Properties, brokerId: Int): Seq[Int] = {
+    val configValue = topicConfig.get(LogConfig.ThrottledReplicasListProp).toString.trim
+    ThrottledReplicaValidator.ensureValid(LogConfig.ThrottledReplicasListProp, configValue)
+    if (configValue.trim == "*")
+      allReplicas
+    else if (configValue.trim == "")
+      Seq()
+    else {
+      configValue.trim
+        .split(":")
+        .map(_.split("-"))
+        .filter(_(1).toInt == brokerId //Filter this replica
+        )
+        .map(_(0).toInt).toSeq //convert to list of partition ids
+    }
```

Comment:
Maybe something like:

```scala
configValue.trim match {
  case "" => ...
  case "*" => ...
  case _ => ...
}
```

I think you also need to handle errors if the format doesn't match what you expect, right? At the moment, we will get unhelpful `ArrayOutOfBoundsException`s and `NumberFormatException`s.